src/main/java/mate/academy/bookshop/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+package mate.academy.bookshop.service.order;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.order.OrderDto;
+import mate.academy.bookshop.dto.orderitem.OrderItemDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.OrderItemMapper;
+import mate.academy.bookshop.mapper.OrderMapper;
+import mate.academy.bookshop.model.Order;
+import mate.academy.bookshop.model.OrderItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.User;
+import mate.academy.bookshop.model.enums.Status;
+import mate.academy.bookshop.repository.cartitem.CartItemRepository;
+import mate.academy.bookshop.repository.order.OrderRepository;
+import mate.academy.bookshop.repository.orderitem.OrderItemRepository;
+import mate.academy.bookshop.repository.shoppingcart.ShoppingCartRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public OrderDto createOrder(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCartByUserId = shoppingCartRepository
+                .findShoppingCartByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shoppingCart by userId: " + user.getId()));
+        Order order = new Order();
+        order.setOrderDate(LocalDateTime.now());
+        order.setUser(user);
+        order.setStatus(Status.PENDING);
+        order.setShippingAddress(user.getShippingAddress());
```

Comment:
As for me, you have long method, you can divide order logic to separate method like createOrder() or buildOrder()